File: Data Structures & Algorithms/validate-parentheses/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        const stack = [];
        const map = {
            ")" : "(",
            "}" : "{",
            "]" : "["
        }
       for(let ch of s) {
        if(ch === '(' || ch === '{' || ch === '['){
            stack.push(ch);
        }
        else{
            if(stack.length > 0 && stack[stack.length-1] === map[ch]){
                stack.pop();
            }
            else {return false}
        }
       }// loop ends here 
       return stack.length === 0 ? "true" : "false";
    }
}

File: Data Structures & Algorithms/is-anagram/submission-0.java
class Solution {
    public boolean isAnagram(String s, String t) {
      HashMap<Character,Integer>mp = new HashMap<>();
      for(char ch : s.toCharArray()){
        mp.put(ch,mp.getOrDefault(ch,0)+1);
      }
      for(char ch : t.toCharArray()){
        if(mp.containsKey(ch)){
            int freq = mp.get(ch)-1;
            if(freq == 0) {
                mp.remove(ch);// remove the ch only whos freq = 0
            }else {// if the freq is not 0 then dec until it becomes -> 0
              mp.put(ch,freq);// update the map with new freq again
            }
        } else {
            return false; //as if the ch is not found in String s  
        }
        } // loop ends here 
     return mp.size() == 0;
     
    }
}
